ModuleThreeFirstTaskConsole/FileSystemVisitor.cs

```diff
@@ -0,0 +1,196 @@
+﻿// <copyright file="FileSystemVisitor.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+
+namespace ModuleThreeFirstTaskConsole
+{
+    /// <summary>
+    /// Can read filesystem and show documents and packages in linear form.
+    /// </summary>
+    public class FileSystemVisitor
+    {
+        private const string STANDARDPATH = @"C:\";
+
+        private readonly IFileSystem _fileSystem;
+        private readonly Func<IFileSystemInfo, bool> _filter;
+        private readonly string _initialPath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileSystemVisitor"/> class.
+        /// </summary>
+        /// <param name="fileSystem">Files and directories.</param>
+        /// <param name="filter">Filter of files and directories.</param>
+        /// <param name="initialPath">Place in filesystem where to start searching.</param>
+        public FileSystemVisitor(
+            IFileSystem fileSystem,
+            Func<IFileSystemInfo, bool> filter,
+            string initialPath)
+        {
+            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            _filter = filter is null ? x => true : filter;
+            _initialPath = initialPath ?? STANDARDPATH;
+            _initialPath = _fileSystem.Directory.Exists(initialPath) ? initialPath : throw new DirectoryNotFoundException("You entered path that doesn't exist.");
+            Stoped = true;
+        }
+
+        /// <summary>
+        /// Events that happen at the begin of search.
+        /// </summary>
+        public event EventHandler<IFileSystemInfo> SearchStarted;
```

Comment:
`EventHandler` is deprecated. Use `Action<T>` instead.

Reply:
Fixed in a883af4